File: abc/LockerDoors/src/Lambda.js
"use strict"

const bruteForce = (n) => {
    const arr = new Array(n)

    arr.fill(false)

    for (let i = 1; i <= n; i++) {
        for (let j = i - 1; j < n; j += i) {
            arr[j] = !arr[j]
        }
    }

    return arr
}

const process = (n) => {
    const arr = Array(n)

    let temp = 0
    for (let i = 1; i <= n; i++) {
        temp = Math.floor(Math.sqrt(i))
        arr[i - 1] = temp * temp === i
    }

    return arr
}

if (module === require.main) {
    const N = 100

    const arr = process(N)

    for (let i = 0; i < N; i++) {
        if (arr[i]) {
            console.log("%d\t  OPEN", i + 1)
        }
    }
}

File: abc/LockerDoors/src/Lambda.c
#include <math.h>
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

bool *bruteForce(int n) {
    bool *arr = (bool *) malloc(n * sizeof(bool));

    memset(arr, false, n);

    for (int i = 1; i <= n; i++) {
        for (int j = i - 1; j < n; j += i) {
            arr[j] = !arr[j];
        }
    }

    return arr;
}

bool *process(int n) {
    bool *arr = (bool *) malloc(n * sizeof(bool));

    int temp;
    for (int i = 1; i <= n; i++) {
        temp = (int) sqrt((double) i);
        arr[i - 1] = temp * temp == i;
    }

    return arr;
}

int main(void) {
    const int N = 100;

    bool *arr = process(N);

    for (int i = 0; i < N; i++) {
        if (arr[i]) {
            printf("%d\t  OPEN\n", i + 1);
        }
    }
}
